components/switch/test/switch-visibility.vdiff.js

```diff
@@ -0,0 +1,152 @@
+	const conditionsMarkup = conditions ? html`
+		These are some conditions that must be met for the activity to be visible.
+		<ul>
+			<li> Condition 1 </li>
+			<li> Condition 2 </li>
+			<li> Condition 3 </li>
+		</ul>
+	` : nothing;
+	return html`
+		<d2l-switch-visibility
+			?on="${on}"
+			text="${ifDefined(text)}"
+			text-position="${ifDefined(textPosition)}">${conditionsMarkup}</d2l-switch-visibility> 
+	`;
+}
+
+describe('d2l-switch-visibility', () => {
+
+	['ltr', 'rtl'].forEach(dir => {
+
+		describe(dir, () => {
+
+			[
+				{ name: 'off', template: create() },
+				{ name: 'on', template: create({ on: true }) },
+				{ name: 'off with conditions', template: create({ conditions: true }) },
+				{ name: 'on with conditions', template: create({ conditions: true, on: true }) },
+				{
+					name: 'on with conditions text position start',
+					template: create({ conditions: true, on: true, textPosition: 'start' })
+				},
+				{
+					name: 'on text-position hidden',
+					template: create({ on: true, textPosition: 'hidden' })
+				},
+				{
+					name: 'on with conditions text-position hidden',
+					template: create({ conditions: true, on: true, textPosition: 'hidden' })
+				},
+				{
+					name: 'off text overridden',
+					template: create({ text: 'Label text has been overridden.' })
+				},
+				{
+					name: 'on text overridden',
+					template: create({ on: true, text: 'Label text has been overridden.' })
+				},
+				{
+					name: 'off with conditons text overridden',
+					template: create({ conditions: true, text: 'Label text has been overridden.' })
+				},
+				{
+					name: 'on with conditions text overridden',
+					template: create({ conditions: true, on: true, text: 'Label text has been overridden.' })
+				},
+				{
+					name: 'on with conditions text-position start text overridden',
+					template: create({ conditions: true, on: true, text: 'Label text has been overridden.', textPosition: 'start' })
+				},
+				{
+					name: 'on text-position hidden text overridden',
+					template: create({ on: true, text: 'Label text has been overridden.', textPosition: 'hidden' })
+				},
+				{
+					name: 'on with conditions text-position hidden text overridden',
+					template: create({ conditions: true, on: true, text: 'Label text has been overridden.', textPosition: 'hidden' })
+				},
+				{
+					name: 'off with only whitespace conditions',
+					template: html`
+						<d2l-switch-visibility>
+						</d2l-switch-visibility>
+					`
+				},
+				{
+					name: 'on with only whitespace conditions',
+					template: html`
+						<d2l-switch-visibility on>
+						</d2l-switch-visibility>
+					`
+				},
+				{
+					name: 'on with only lots of whitespace conditions',
+					template: html`
+						<d2l-switch-visibility on>
+											
+						</d2l-switch-visibility> 
+					`
+				},
+				{
+					name: 'on with only loooots of whitespace conditions',
```

Comment:
Minor: Lol what is this testing, that the whitespace doesn't show up? Wondering if you could do something like https://github.com/BrightspaceUI/core/blob/15d79588871270849320a3928fa40cdeed95ede3/components/button/test/floating-buttons.vdiff.js#L5

Reply:
Haha I think it's there because when there are "real" non-whitespace things in the slot it renders the help tooltip, so this is testing that they're ignoring empty text nodes in the slot. But yeah, I'll try changing it to something less verbose.

Reply:
Alright, I've combined the "whitespace" tests into one and am using that `Array(blah).fill` method to fill in various types of whitespace.

Reply:
Did you mean to remove the off case too? If so probably need to delete that golden, or else it'll be deleted with the next failing vdiff run

Reply:
I did mean to remove them but missed the goldens, thanks -- fixed.